viewer/core/static/js/app/components/containers/app-container.js

```diff
@@ -93,8 +117,9 @@ class AppContainer extends React.Component {
 
 const mapStateToProps = function(store, ownProps) {
   return {
-    subgroups: store.subgroupsState.subgroups,
+    datasets: store.datasetState.datasets,
     metricMetadata: store.metricMetadataState.metadata,
+    subgroups: store.subgroupsState.subgroups,
```

Comment:
I was thinking of re-doing the subgroup API stuff to be more like your dataset approach but you're absolutely right that we don't need both.